lib/miq_expression/target.rb

```diff
@@ -0,0 +1,67 @@
+class MiqExpression::Target
+  ParseError = Class.new(StandardError)
+
+  def self.parse!(field)
+    parse(field) || raise(ParseError, field)
+  end
+
+  # example .parse('Host.vms-host_name')
+  # returns hash:
+  # {:model_name => Host<ApplicationRecord> , :associations => ['vms']<Array>, :column_name => 'host_name' <String>}
+  def self.parse_params(field)
+    match = self::REGEX.match(field) || return
+    # convert matches to hash to format
+    # {:model_name => 'User', :associations => ...}
```

Comment:
thanks for comment. cool implementation, but yea, it needed a comment.

in general this method looks very complicated to me. :(